packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -455,10 +617,28 @@ class Context {
       '--DartDefines=${environment['DART_DEFINES'] ?? ''}',
       '--ExtraFrontEndOptions=${environment['EXTRA_FRONT_END_OPTIONS'] ?? ''}',
       '-dSrcRoot=${environment['SRCROOT'] ?? ''}',
-      '-dTargetDeviceOSVersion=${environment['TARGET_DEVICE_OS_VERSION'] ?? ''}',
       '-dDevDependenciesEnabled=${environment['FLUTTER_DEV_DEPENDENCIES_ENABLED'] ?? ''}',
     ]);
 
+    if (platform == TargetPlatform.ios) {
+      flutterArgs.add('-dTargetDeviceOSVersion=${environment['TARGET_DEVICE_OS_VERSION'] ?? ''}');
+      final String? expandedCodeSignIdentity = environment['EXPANDED_CODE_SIGN_IDENTITY'];
+      if (expandedCodeSignIdentity != null &&
+          expandedCodeSignIdentity.isNotEmpty &&
+          environment['CODE_SIGNING_REQUIRED'] != 'NO') {
+        flutterArgs.add('-dCodesignIdentity=$expandedCodeSignIdentity');
+      }
+    }
+    if (platform == TargetPlatform.macos && command == 'build') {
+      final String ephemeralDirectory = '$sourceRoot/Flutter/ephemeral';
+      final String buildInputsPath = '$ephemeralDirectory/FlutterInputs.xcfilelist';
+      final String buildOutputsPath = '$ephemeralDirectory/FlutterOutputs.xcfilelist';
```

Comment:
[Drives me nuts how different the input/output fingerprinting is between iOS and macOS]